app/controllers/groups_controller.rb

```diff
@@ -362,16 +412,27 @@ def request_membership
     group = find_group(:id)
     group_name = group.name
 
+    GroupRequest.create!(group: group, user: current_user, reason: params[:reason])
```

Comment:
I think this should probably happen in the same transaction as `PostCreate.create!`

Reply:
There is a transaction inside PostCreator and it is not something I would want to pollute with group requests.

Reply:
O I don’t mean that it should use the transaction inside. I meant that it should happen in the same transaction which you can wrap around post creator